src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,116 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getPerson)
+                .map(Person::getAge)
+                .reduce(0, (a, b) -> a + b)
+                .doubleValue() / employees.size();
+
+        assertEquals(33.66, expected, 0.1);
+    }
+
+    @Test
+    public void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(Comparator.comparing(p -> p.getFullName().length()))
+                .get();
+
+        assertEquals(expected, employees.get(1).getPerson());
+    }
+
+    @Test
+    public void findEmployeeWithMaximumDurationAtOnePosition() {
+        List<Employee> employees = getEmployees();
+
+        Employee expected = employees.stream()
+                .max(Comparator.comparing(Employee::getJobHistory, Comparator.comparing(job -> job.stream()
+                        .map(JobHistoryEntry::getDuration).max(Integer::compareTo).orElseGet(() -> 0))))
+                .get();
+
+        assertEquals(expected, employees.get(4));
+    }
+
+    /**
+     * Вычислить общую сумму заработной платы для сотрудников.
+     * Базовая ставка каждого сотрудника составляет 75_000.
+     * Если на текущей позиции (последняя в списке) он работает больше трех лет - ставка увеличивается на 20%
+     */
+    @Test
+    public void calcTotalSalaryWithCoefficientWorkExperience() {
+        List<Employee> employees = getEmployees();
+
+        final int COMMON_PRICE = 75_000;
+        final double COEFFICIENT = 1.2;
+
+        Double expected = employees.stream()
+                .map(Employee::getJobHistory)
+                .filter(jobs -> jobs.size() > 0)
+                .map(jobs -> jobs.get(jobs.size() - 1))
+                .map(job -> job.getDuration() > 3 ? COMMON_PRICE * COEFFICIENT : (double) COMMON_PRICE)
```

Comment:
I like that you use constants instead of "magic numbers", but in that case you must extract "3" too.